File: python/test/videolibrary_test.py
from src.video_library import VideoLibrary


def test_library_has_all_videos():
    library = VideoLibrary()
    assert len(library.get_all_videos()) == 5


def test_parses_tags_correctly():
    library = VideoLibrary()
    video = library.get_video("amazing_cats_video_id")

    assert video is not None
    assert video.title == "Amazing Cats"
    assert video.video_id == "amazing_cats_video_id"
    assert set(video.tags) == {"#cat", "#animal"}


def test_parses_video_correctly_without_tags():
    library = VideoLibrary()
    video = library.get_video("nothing_video_id")

    assert video is not None
    assert video.title == "Video about nothing"
    assert video.video_id == "nothing_video_id"
    assert video.tags == ()

File: solutions/cpp/solution1/test/videolibrary_test.cpp
#include "../src/videolibrary.h"

#include <gmock/gmock.h>
#include <gtest/gtest.h>

#include "../src/video.h"

using ::testing::ContainsRegex;
using ::testing::HasSubstr;

TEST(VideoLibrary, VideoLibraryHasAllVideos) {
  VideoLibrary videoLibrary = VideoLibrary();
  EXPECT_EQ(videoLibrary.getVideos().size(), 5);
}

TEST(VideoLibrary, VideoLibraryParsesTagsCorrectly) {
  std::vector<std::string> tags{"#cat", "#animal"};
  VideoLibrary videoLibrary = VideoLibrary();
  const Video *video = videoLibrary.getVideo("amazing_cats_video_id");
  bool result = std::equal(tags.begin(), tags.end(), video->getTags().begin());
  EXPECT_EQ("Amazing Cats", video->getTitle());
  EXPECT_EQ("amazing_cats_video_id", video->getVideoId());
  EXPECT_TRUE(result);
}

TEST(VideoLibrary, testLibraryParsesVideoCorrectlyWithoutTags) {
  VideoLibrary videoLibrary = VideoLibrary();
  const Video *video = videoLibrary.getVideo("nothing_video_id");
  EXPECT_EQ("Video about nothing", video->getTitle());
  EXPECT_EQ("nothing_video_id", video->getVideoId());
  EXPECT_TRUE(video->getTags().empty());
}
